packages/vue-components/src/panels/PanelBase.js

```diff
@@ -95,80 +95,133 @@ export default {
+        // Only preload, do not expand the panel.
+        return;
+      }
+      // Don't play the transition for this case as the loading should feel 'instant'.
       if (this.expandedBool) {
         this.$refs.panel.style.maxHeight = 'none';
         return;
       }
 
-      /*
-      Otherwise, since the vue transition is dependent on localExpanded, we have to manually
-      set our own transition end handlers here for the initial loading of the content.
-      */
-      const onExpandDone = () => {
-        this.$refs.panel.style.maxHeight = 'none';
-        this.$refs.panel.removeEventListener('transitionend', onExpandDone);
-      };
-
-      this.$refs.panel.addEventListener('transitionend', onExpandDone);
+      // For expansion transition to 'continue' after src is loaded.
       this.$refs.panel.style.maxHeight = `${this.$refs.panel.scrollHeight}px`;
     },
-    beforeExpand(el) {
-      el.style.maxHeight = '0';
-    },
-    duringExpand(el) {
-      jQuery('html').stop();
-      el.style.maxHeight = `${el.scrollHeight}px`;
-    },
-    afterExpand(el) {
-      el.style.maxHeight = 'none';
-    },
-    beforeCollapse(el) {
-      el.style.maxHeight = `${el.scrollHeight}px`;
+    initCollapsedPanelHeight() {
+      // To set panel's default collapsed panel height.
+      this.collapsedPanelHeight = 0;
     },
-    duringCollapse(el) {
-      if (this.$el.getBoundingClientRect().top < 0) {
-        jQuery('html').animate({
-          scrollTop: window.scrollY + this.$el.getBoundingClientRect().top - 3,
-        }, 500, 'swing');
+    initPanel() {
+      this.$refs.panel.addEventListener('transitionend', () => {
+        /*
+          If the panel's content grows, its height should accomodate it accordingly (e.g. nested panels).
+          We will always set the panel's maxHeight to 'none' after expansion.
+          This is to prevent maxHeight (which is used for transition) from restricting panel's growth.
+
+          At the end of first expansion transition, if src has not finished loading, then we should not
+          set panel's maxHeight to none.
+
+          This is because once src finishes loading, retriever component will emit an event to
+          set the flag isRetrieverLoadDone to true and update the maxHeight again.
+
+          This will trigger a second expansion transition. And when this happens, we need the original
+          maxHeight (from first expansion transition) to ensure the transition can 'continue' smoothly.
+
+          Once that 'second' transition is finished, the flag isRetrieverLoadDone would have already
+          been set to true and the end of the 'second' transition will trigger this event listener
+          again to set maxHeight to none.
+        */
+        if (this.localExpanded && this.isRetrieverLoadDone) {
+          this.$refs.panel.style.maxHeight = 'none';
+        }
+      });
+
+      this.wasRetrieverLoaded = this.localExpanded;
```

Comment:
could move this to `created()` too

Reply:
Hmm yeah you're right. I was still holding onto some old assumptions that this needs to be here. But with the new revisions according to your review, this can be shifted to created().